engine/src/main/java/com/datatorrent/stram/stream/BufferServerSubscriber.java

```diff
@@ -242,6 +243,21 @@ public long getByteCount(boolean reset)
     }
 
     @Override
+    public int size(final boolean dataTupleAware)
+    {
+      int size = size();
+      if (dataTupleAware) {
```

Comment:
@gauravgopi123 Can you please describe use case? As for SpscArrayQueue size() is an estimate, accounting for control tuples does not provide any better estimate.